File: Source/Geometry/polygon_area.c
#include <stdlib.h>

typedef struct {
    double x;
    double y;
} Point;

double calculatePolygonArea(Point* vertices, int numVertices) {
    double area = 0.0;
    int j = numVertices - 1;

    for (int i = 0; i < numVertices; i++) {
        area += (vertices[j].x + vertices[i].x) * (vertices[j].y - vertices[i].y);
        j = i;
    }

    return abs(area / 2.0);
}

File: Source/Geometry/polygon_area.js
function calculatePolygonArea(vertices) {
    let area = 0;
    let j = vertices.length - 1;

    for (let i = 0; i < vertices.length; i++) {
        area += (vertices[j].x + vertices[i].x) * (vertices[j].y - vertices[i].y);
        j = i;
    }

    return Math.abs(area / 2);
}
